services/quizzes/src/components/SubmissionComponents/MultipleChoice.tsx

```diff
@@ -157,60 +160,74 @@ const MultipleChoiceSubmission: React.FC<QuizItemSubmissionComponentProps> = ({
 }
 
 interface MultipleChoiceDirectionProps {
-  submissionFeedback: QuizItemOption | undefined
+  feedbackDisplayPolicyAccordingToModelSolution:
+    | "DisplayFeedbackOnQuizItem"
+    | "DisplayFeedbackOnAllOptions"
+    | undefined
+  submissionFeedback: string | undefined
   selectedAnswer: boolean
-  feedbackDisplayPolicy: "DisplayFeedbackOnQuizItem" | "DisplayFeedbackOnAllOptions" | undefined
+  modelSolutionFeedback: QuizItemOption | undefined
 }
 
 const RowSubmissionFeedback: React.FC<MultipleChoiceDirectionProps> = ({
+  feedbackDisplayPolicyAccordingToModelSolution,
   submissionFeedback,
   selectedAnswer,
-  feedbackDisplayPolicy,
-}) => (
-  <div>
-    {feedbackDisplayPolicy === "DisplayFeedbackOnQuizItem" && submissionFeedback ? (
-      <>
-        {selectedAnswer ? (
-          <div
-            className={css`
-              margin-left: 0.5em;
-              display: flex;
-              border-left: ${submissionFeedback.correct
-                ? `6px solid #1F6964`
-                : `6px solid #A84835`};
-              box-sizing: border-box;
-              padding: 0.5rem 0px 0.5rem 0.5rem;
-              margin-bottom: 5px !important;
-            `}
-          >
-            <p>
-              {submissionFeedback.correct
-                ? submissionFeedback.successMessage
-                : submissionFeedback.failureMessage}
-            </p>
-          </div>
-        ) : null}
-      </>
-    ) : null}
-    {feedbackDisplayPolicy === "DisplayFeedbackOnAllOptions" && submissionFeedback ? (
-      <div
-        className={css`
-          margin-left: 0.5em;
-          display: flex;
-          border-left: ${submissionFeedback.correct ? `6px solid #1F6964` : `6px solid #A84835`};
-          box-sizing: border-box;
-          padding: 0.5rem 0px 0.5rem 0.5rem;
-          margin-bottom: 5px !important;
-        `}
-      >
-        <p>
-          {submissionFeedback.correct
-            ? submissionFeedback.successMessage
-            : submissionFeedback.failureMessage}
-        </p>
-      </div>
-    ) : null}
-  </div>
-)
+  modelSolutionFeedback,
+}) => {
+  return (
+    <div>
+      {feedbackDisplayPolicyAccordingToModelSolution === "DisplayFeedbackOnQuizItem" ? (
+        <>
+          {modelSolutionFeedback && selectedAnswer ? (
+            <div
+              className={css`
+                margin-left: 0.5em;
+                display: flex;
+                border-left: ${modelSolutionFeedback.correct
+                  ? `6px solid #1F6964`
+                  : `6px solid #A84835`};
+                box-sizing: border-box;
+                padding: 0.5rem 0px 0.5rem 0.5rem;
+                margin-bottom: 5px !important;
+              `}
+            >
+              <p>
+                {modelSolutionFeedback.correct
```

Comment:
Cause in my mind this would use model solution if it's available, and if not it would try to use feedbackForThisOption. I just don't understand why this is working.